supercommand.go

```diff
@@ -526,6 +535,18 @@ func (c *SuperCommand) Run(ctx *Context) error {
 	return err
 }
 
+func (c *SuperCommand) parseFormattingFlag() {
+
+	// Check if they are already global registered
+	if c.flags.Lookup("format") != nil {
+		c.formattingType = c.flags.Lookup("format").Value.String()
+	} else {
+		c.flags.StringVar(&c.formattingType, "format", "", "")
+		_ = c.flags.Parse(true, c.flags.Args())
```

Comment:
Why is it safe to do this? I'm assuming that if this fails we can just fallback to format `""`. We should seriously think about this, as I'm unsure what the actual correct response here is. I'm probably going to guess if this happens, we should return an error, but in which type, as we can't tell which one you asked for.

Chicken and egg problem :D